echidna/core/dsextract.py

```diff
@@ -0,0 +1,608 @@
+import numpy
+import rat
+import echidna.calc.constants as const
+
+
+def function_factory(dimension, **kwargs):
+    '''Factory function that returns a dsextract class
+    corresponding to the dimension (i.e. DS parameter)
+    that is being extracted.
+
+    Args:
+      dimension (str): to extract from a RAT DS/ntuple file.
+      \**kwargs (dict): to be passed to and checked by the extractor.
```

Comment:
@jwaterfield not sure what the `\**` is doing here. `kwargs` is currently not displayed in docs. See screenshot.

![screenshot from 2015-09-07 12 33 42](https://cloud.githubusercontent.com/assets/1931666/9715908/f63e4312-555c-11e5-9217-b2db62ad7f65.png)